Add cool useInView hook 💪

JustSuperHuman opened the pull request:
Thought this was a really cool / light hook that would fit in here! Added the example / doc / and hook to the repository! 

Hope it fits 😄
    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Added a new useInView hook to detect when an element enters the viewport, with options for margin, amount, and one-time detection.

- **New Features**
  - useInView hook for tracking element visibility using Intersection Observer.
  - Example demo and documentation included.

<!-- End of auto-generated description by cubic. -->

JustSuperHuman committed:
Add cool useInView hook 💪

JustSuperHuman committed:
Update registry/examples/use-in-view-demo.tsx

Good bot

Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

JustSuperHuman commented on registry/hooks/use-in-view.ts:
```diff
@@ -0,0 +1,112 @@
+	const wasInViewRef = useRef(false)
+
+	useEffect(() => {
+		if (!ref || !ref.current) return
```
Can you fix this Cubit bot :P

strlrd-29 commented:
Hey @JustSuperHuman, thanks for adding this awesome hook! I’ll review the PR later and get it merged.

strlrd-29 commented:
Hey @JustSuperHuman, I tried running your branch locally, but I keep running into the error shown in the screenshot below:

<img width="1321" alt="image" src="https://github.com/user-attachments/assets/e1a48030-38c3-4bb0-9dbe-f4155297cfc4" />

Is there something I might be missing on my end? Are you seeing the same issue?

Also, could you please update `/config/docs.ts` to include the hook in the `sidebarNav`? Really appreciate it 🙏

JustSuperHuman committed:
fix: useInView index / docs / example

JustSuperHuman commented:
Thanks so much for the review! I added the link to the sidebar and fixed the example component not showing :) 

I'm not getting the MDX build error tho 🤔

Using Windows. Tried in WSL also and getting good output. 

![image](https://github.com/user-attachments/assets/6f8a2c54-9d29-4081-950c-97e8968cb2d8)

JustSuperHuman commented:
I get issues with NPM if I don't run with `npm i --legacy-peer-deps` but don't think that's related.
